### ISSUE TITLE
Exception thrown when trying to check if index exists

### ISSUE THREAD

**Issue by oliv0533** [elastic/elasticsearch-net#5730]
**NEST/Elasticsearch.Net version**:
7.13.0

**Elasticsearch version**:
7.13.1

**Description of the problem including expected versus actual behavior**:
We have 3 clusters. 2 in Elastic Cloud and one running locally in docker. When trying to determine if an index exists, using ExistsAsync(), we get a BadResponse:

```
# FailureReason: BadResponse while attempting HEAD on https://elastic:SdHPdB7XrHnNh4hMZNZAdpoa@8a4d49ea484146418345bf750e01b415.eu-west-1.aws.found.io:9243/_alias/indexer_rigel_pickuppoints_coolrunnerdaohome-alias?pretty=true&error_trace=true
# Audit trail of this API call:
 - [1] BadResponse: Node: https://elastic:redacted@8a4d49ea484146418345bf750e01b415.eu-west-1.aws.found.io:9243/ Took: 00:00:00.0782374
# OriginalException: Elasticsearch.Net.ElasticsearchClientException: Request failed to execute. Call: Status code 200 from: HEAD /_alias/indexer_rigel_pickuppoints_coolrunnerdaohome-alias?pretty=true&error_trace=true
   at Elasticsearch.Net.Transport`1.HandleElasticsearchClientException(RequestData data, Exception clientException, IElasticsearchResponse response)
   at Elasticsearch.Net.Transport`1.FinalizeResponse[TResponse](RequestData requestData, IRequestPipeline pipeline, List`1 seenExceptions, TResponse response)
   at Elasticsearch.Net.Transport`1.RequestAsync[TResponse](HttpMethod method, String path, CancellationToken cancellationToken, PostData data, IRequestParameters requestParameters)
   at Reshopper.BusinessLogic.Infrastructure.Elasticsearch.ElasticsearchHelper.<>c__DisplayClass5_0.<<GetWhichIndexNamesAliasPointsToAsync>b__0>d.MoveNext() in C:\Source\Reshopper\backend\src\Reshopper.BusinessLogic\Infrastructure\Elasticsearch\ElasticsearchHelper.cs:line 74
# Request:
<Request stream not captured or already read to completion by serializer. Set DisableDirectStreaming() on ConnectionSettings to force it to be set on the response.>
# Response:

# TCP states:
  Established: 163
  TimeWait: 11
  CloseWait: 2

# ThreadPool statistics:
  Worker: 
    Busy: 2
    Free: 32765
    Min: 8
    Max: 32767
  IOCP: 
    Busy: 1
    Free: 999
    Min: 8
    Max: 1000

# Exception:
Elasticsearch.Net.ElasticsearchClientException: Request failed to execute. Call: Status code 200 from: HEAD /_alias/indexer_rigel_pickuppoints_coolrunnerdaohome-alias?pretty=true&error_trace=true
   at Elasticsearch.Net.Transport`1.HandleElasticsearchClientException(RequestData data, Exception clientException, IElasticsearchResponse response)
   at Elasticsearch.Net.Transport`1.FinalizeResponse[TResponse](RequestData requestData, IRequestPipeline pipeline, List`1 seenExceptions, TResponse response)
   at Elasticsearch.Net.Transport`1.RequestAsync[TResponse](HttpMethod method, String path, CancellationToken cancellationToken, PostData data, IRequestParameters requestParameters)
   at Reshopper.BusinessLogic.Infrastructure.Elasticsearch.ElasticsearchHelper.<>c__DisplayClass5_0.<<GetWhichIndexNamesAliasPointsToAsync>b__0>d.MoveNext() in C:\Source\Reshopper\backend\src\Reshopper.BusinessLogic\Infrastructure\Elasticsearch\ElasticsearchHelper.cs:line 74
```
We have also tried with a missing index and get a 404 as expected, but it still throws the exception and the "Exists" property is not present on the exception. 
This only happens on one of the cloud clusters, the other 2 clusters works as expected with the "Exists" property available even when a 404 is returned.

When doing the exact same calls using postman, it returns the status codes as it should. 

It seems one of our clusters triggers something in the client which makes it throw an exception instead of handling the 404 and 200 the expected way.

The same thing happens when trying to check if an alias exists using AliasExistsAsync(). Exact same behavior

**Steps to reproduce**:
No clue, Only happens on one cluster out of 3.

**Expected behavior**
No exceptions throw and the possibility to check the exists variable in the answer from the cluster, instead of a thrown exception.

**Comment by stevejgordon** [comment]
Hi, @oliv0533. Thanks for reporting this. I have reproduced the issue and I'm looking at a client-side fix around response validation. Could you also please let me know which cloud provider and region both cloud deployments are in and which one succeeds and which fails?

**Comment by stevejgordon** [comment]
@oliv0533 - We isolated this to a change in 7.13.0 which affects these requests against Elastic Cloud deployments. I have released 7.13.1 which includes a fix for this issue from #5735.

**Comment by oliv0533** [comment]
@stevejgordon  Thanks for the swift action.
